### PR TITLE
hermes client explicit Avro support

### PR DESCRIPTION
**Pull request by adamdubiel**
This PR adds explicit JSON and AVRO support in Hermes Client. I also refactored tests to Groovy to take advantage of advanced parametrization when testing multiple sender implementations.

### CODE REVIEW

File: docs/docs/user/java-client.md [hunk 1]
```diff
@@ -40,6 +40,55 @@ HermesClientBuilder.hermesClient(...)
...
+
+```java
+hermesClient.publishJSON("com.group.json", "{hello: 1}");
+```
+
+Avro sender sets `avro/binary` content type. It also requires to pass Avro schema version of this message, which is
+passed on to Hermes in `Schema-Version` header.
+
+```java
+hermesClient.publisAvro("com.group.avro", 1, avroMessage.getBytes());
```
Comment (druminski) [inline docs/docs/user/java-client.md]: publish*

File: hermes-client/src/main/java/pl/allegro/tech/hermes/client/HermesClient.java [hunk 2]
```diff
@@ -10,6 +10,10 @@
 import static java.util.stream.IntStream.range;
 
 public class HermesClient {
+
+    private static final String APPLICATION_JSON = "application/json;charset=UTF-8";
+    private static final String AVRO_BINARY = "avro/binary;charset=UTF-8";
```
Comment (druminski) [inline hermes-client/src/main/java/pl/allegro/tech/hermes/client/HermesClient.java]: hmm do we need set charset if we send binary data?

File: hermes-client/src/test/groovy/pl/allegro/tech/hermes/client/HermesClientTesty.groovy [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+
+import java.util.concurrent.CompletableFuture
+import java.util.concurrent.CountDownLatch
+import java.util.function.Supplier
+
+import static java.net.URI.create
+import static java.util.concurrent.CompletableFuture.completedFuture
+import static pl.allegro.tech.hermes.client.HermesClientBuilder.hermesClient
+
+class HermesClientTesty extends Specification {
```
Comment (druminski) [inline hermes-client/src/test/groovy/pl/allegro/tech/hermes/client/HermesClientTesty.groovy]: Tests*

**Comment by druminski** [comment]
:+1:

File: hermes-client/src/main/java/pl/allegro/tech/hermes/client/HermesClient.java (hunk 2 above)
Comment (adamdubiel) [inline hermes-client/src/main/java/pl/allegro/tech/hermes/client/HermesClient.java]: good remark, we should not

**Comment by wojtkiewicz** [comment]
:+1:

**Pull request merged**